src/QMCDrivers/WFOpt/EngineHandle.h

```diff
@@ -38,19 +38,21 @@ class EngineHandle
    *
    *\param[in] num_params           Number of optimizable parameters
    */
-  virtual void prepareSampling(int num_params) = 0;
+  virtual void prepareSampling(int num_params,int num_samples) = 0;
   /** Function for passing derivative ratios to optimizer engines
    *
    * \param[in] energy_list         Vector of local energy values
    * \param[in] dlogpsi_array       Parameter derivatives of log psi
    * \param[in] dhpsioverpsi_array  Parameter derivatives of local energy
-   * \param[in] ib                  Sample index
+   * \param[in] crowd_id            Crowd number
+   * \param[in] local_index         Crowd local index
+   * \param[in] index_offset        Offset for sample's index
```

Comment:
I'm not sure this is best described as an offset. Maybe "sample_index"?  This is the index of the sample on this MPI rank.   The extent of the index is [0, rank_local_num_samples_), so maybe rank_local_sample_index?  The rank_local part is a bit wordy, so sample_index should be sufficient.

Reply:
Ok, I have changed the naming to sample_index to make things clearer.